src/viewport.js

```diff
@@ -940,8 +942,23 @@ $.Viewport.prototype = {
             this.degreesSpring.isAtTargetValue()) {
             return this;
         }
+       this.rotationPivot = pivot;
         if (immediately) {
-            this.degreesSpring.resetTo(degrees);
+            if(pivot){
+                var changeInDegrees = degrees - this._oldDegrees;
+                if(!changeInDegrees){
+                    return this;
+                }
+                var delta = pivot.minus(this.getCenter());
```

Comment:
This code looks very similar to the version in `update`. Maybe it's worth having a common routine they both call, so it's not duplicated? The routine should be private, which in this code base just means prefixing the name with `_`.